userscripts/ExTagLink.user.js

```diff
@@ -68,16 +69,53 @@ for (const com of $$('.c6')) {
       magnetLinks.push(magnetLink);
     }
 
-    const pixivLink = textNode.textContent.match(/https:\/\/www\.pixiv\.net\/\S+/g);
+    const pixivLink = textNode.textContent.match(
+      /https:\/\/www\.pixiv\.net\/\S+/g
+    );
 
     if (pixivLink) {
       pixivLinks.push(pixivLink);
     }
 
+    /**
+     * To match fanbox URLS like
+     * https://lambda.fanbox.cc/
+     * https://lambda.fanbox.cc/posts/123
+     * https://fanbox.cc/@lambda
+     * https://fanbox.cc/@lambda/posts/123
+     */
+    const fanboxLink = textNode.textContent.match(
+      /(https?:\/\/[\w-]+\.fanbox\.cc\/posts\/\d+|https?:\/\/(?:www\.)?fanbox\.cc\/@[\w-]+\/posts\/\d+|https?:\/\/(?:www\.)?fanbox\.cc\/@[\w-]+|https?:\/\/(?:www\.)?[\w-]+\.fanbox\.cc)/g
+    );
+
+    if (fanboxLink) {
+      fanboxLinks.push(fanboxLink);
+    }
+
     textNode = walker.nextNode();
   }
 }
 
+if (fanboxLinks.length) {
+  const linkItems = fanboxLinks.map((link) => {
+    let name;
+    if (link.includes('fanbox.cc/posts/')) {
+      // match https://lambda.fanbox.cc/posts/123
+      // match https://fanbox.cc/@lambda/posts/123
+      name = link.match(/posts\/(\d+)/)[1];
+    } else if (link.includes('fanbox.cc/@')) {
+      // match https://fanbox.cc/@lambda
+      name = link.match(/@(\S+)/)[1];
+    } else {
+      // match https://lambda.fanbox.cc/
+      name = new URL(link).hostname.match(/([\w-]+)\.fanbox\.cc/)[1];
+    }
+    return { name, link };
```

Comment:
Ok, I think there is a misunderstanding. I'm not arguing about your regex on matching fanbox link.

What I expected was to show the artist name from the following links, but I only see 'www'.

https://exhentai.org/g/2757520/fc42e712c9/
https://exhentai.org/g/2757482/df77cc8161/
https://exhentai.org/g/2757469/2d57af1548/
https://exhentai.org/g/2757413/70f2d16235/

That is why I provided the regex above.

Reply:
I think the following formats are more intuitive:

For links like:

- `https://lambda.fanbox.cc/posts/123`
- `https://fanbox.cc/@lambda/posts/123`

The name could be `lambda:123` or `lambda/123` or `@lambda:123`

For links like:  

- `https://lambda.fanbox.cc/`
- `https://fanbox.cc/@lambda`
- `https://www.fanbox.cc/@lambda`

The name would be `lambda` or `@lambda`